data/transactions/logic/assembler.go

```diff
@@ -1567,41 +1591,62 @@ func (ops *OpStream) assemble(text string) error {
 		if ops.versionedPseudoOps == nil {
 			ops.versionedPseudoOps = prepareVersionedPseudoTable(ops.Version)
 		}
-		opstring := fields[0]
-		if opstring[len(opstring)-1] == ':' {
-			ops.createLabel(opstring[:len(opstring)-1])
-			fields = fields[1:]
-			if len(fields) == 0 {
-				ops.trace("%3d: label only\n", ops.sourceLine)
+		var current []string
+		var next []string
+		next = fields
+		for current, next = processFields(next); len(current) > 0 || len(next) > 0; current, next = processFields(next) {
+			if len(current) == 0 {
 				continue
 			}
-			opstring = fields[0]
-		}
-		spec, expandedName, ok := getSpec(ops, opstring, fields[1:])
-		if ok {
-			ops.trace("%3d: %s\t", ops.sourceLine, opstring)
-			ops.recordSourceLine()
-			if spec.Modes == modeApp {
-				ops.HasStatefulOps = true
+			opstring := current[0]
+			if len(opstring) == 0 {
+				continue
 			}
-			args, returns := spec.Arg.Types, spec.Return.Types
-			if spec.refine != nil {
-				nargs, nreturns := spec.refine(&ops.known, fields[1:])
-				if nargs != nil {
-					args = nargs
-				}
-				if nreturns != nil {
-					returns = nreturns
-				}
+			if strings.HasPrefix(opstring, "//") {
+				// semicolon inside comment is not counted as newline
```

Comment:
Is this the problem this code addresses? It seems to me this handles:
```
int 2; // comment
```
That is, it makes sure that a leading comment after a semi is detected. For the case of a semi inside of a comment, I think/hope fieldsFromLine has already thrown it away.